### DISCUSSION TITLE
RPi4 build with SATA-USB expansion board and 5V fan

### DISCUSSION THREAD

**Discussion by openoms** [raspiblitz/raspiblitz#3938] [Research]
In connection with the metal case (https://diynodes.com/product/zkds-metal-case-for-the-raspiblitz/) I have been experimenting with this SATA-USB expansion board and found it very stable.

Sources:
Amazon:
https://www.amazon.co.uk/Geekworm-Raspberry-Expansion-USB3-0-Shield/dp/B07VXF2HJG/
DIYnodes.com (bitcoin only): https://diynodes.com/product/sata-usb-expansion-board-and-usb-bridge/

It has 2 pairs of 5V power pins too, so there are more options to attach a fan.
This one seems to be good quality, silent and efficient (keeps the temp at 58-62 degrees max with repeated stresstest runs `./config.scripts/blitz.stresstest.sh`):
https://www.ebay.co.uk/itm/Fan-Square-Cooling-Fan-with-Heatsink-Cooler-Kit-For-Raspberry-Pi-4-3-V-df/372980446235
It sits on a heatsink which has a sticky layer on the bottom.
there can be plenty of similar fans be found on amazon/ebay/aliexpress. 
Search for `Square Fan Rasperry Pi`
![photo6019203628495122546](https://user-images.githubusercontent.com/43343391/70638059-1f752400-1c30-11ea-87a2-185f79dbaa96.jpg)
![photo6019203628495122547](https://user-images.githubusercontent.com/43343391/70638062-20a65100-1c30-11ea-8e00-718febf92f82.jpg)

I am also using some 22mm x M2.5 screws (with M2.5 locking nuts) to support the screen from below. This keeps it very stable when pressed.

**Comment by openoms** [comment]
More for the Maximal package. BIg SSd-s seem to draw siginificantly more power than the smaller ones.
500GB SSD-s are taking 1-1.2 Amperes.
1TB I have seen are 1.6-1.7 (and 1.7 was causing problems)

I have tested 3 types of 1 TB SSD-s with this setup so far:

**GOOD**:
SanDisk SSD PLUS 1 TB Sata III 2.5 Inch Internal SSD, Up to 535 MB/s: https://www.amazon.co.uk/SanDisk-SDSSDA-1T00-G26-Internal-Speed-Write/dp/B07D998212
power req: **1.6 A** - no issues so far

WD Blue 3D NAND Internal SSD 2.5 Inch SATA - 1 TB
https://www.amazon.co.uk/gp/product/B073SBQMCX/ref=ppx_yo_dt_b_asin_title_o01_s00?ie=UTF8&psc=1
power req: **1.6 A** - no issues so far
![IMG_20200203_125436](https://user-images.githubusercontent.com/43343391/73675625-b1d2fa80-46aa-11ea-965d-1833f84859ea.jpg)


**BAD**:
Crucial MX500 CT1000MX500SSD1 1 TB (3D NAND, SATA, 2.5 Inch, Internal SSD)
https://www.amazon.co.uk/Crucial-MX500-CT1000MX500SSD1-NAND-Internal/dp/B078211KBB
power req: **1.7 A** - on high load of the node (electrs + txindex indexing at the same time)  had read errors and froze (probably due to the power spike)
![IMG_20200203_112523](https://user-images.githubusercontent.com/43343391/73675647-ba2b3580-46aa-11ea-8187-7b492679e8ac.jpg)


The power was a single 3A Pimoroni PSU from the shopping list for all. The X825 board can be powered separately too with a 5V barrel connector in case that would be needed in the future. Already very useful when cloning the disks.

**Comment by openoms** [comment]
Link to a build with the X825 board, heatsink-case and 22mm * M2.5 screws working together well:
https://twitter.com/DIYnodes/status/1248587471983607808

Fits the metal case too.

**Comment by openoms** [comment]
note from TG regarding the Geekworm Raspberry Pi 4 X872 M.2 NVMe SSD Expansion Board https://www.amazon.co.uk/Geekworm-Raspberry-Expansion-Compatible-Support/dp/B0855DMH6C

https://t.me/raspiblitz/40612
![image](https://user-images.githubusercontent.com/43343391/94683243-1750b700-031e-11eb-91c5-728c1535dfb5.png)

**Comment by maximlomans** [comment]
i had a similar issue with fan power inputs being used by the tft screen - in the end i stripped an old usb cable to power the internal fan i have a second external fan fixed in place temp never exceeds 47* and a samsung t5 1T b specifically designed as a smartphone external drive - flawless

**Comment by marssystems** [comment]
Use this board from Geekworm and you won't need the SD card any more.

https://geekworm.com/products/x883

You will also need one of these to connect power to the RPi -

https://geekworm.com/products/gpio-1-to-2-extender
